From 712bb507a0656bfb82465b7cd2eedd6efde3e10c Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Mon, 3 Sep 2018 09:57:26 +0200
Subject: [PATCH 35/52] PhoneWindowManager: Check if proposed rotation is in
 range

* Some OEMs *coughs OnePlus* decided it was a great idea to add a fourth kind of rotation ( flat )
* Of course Android doesn't know how to handle it, thus crashing in some usecases

Change-Id: I9c64adcbe557ce2954989af5d0beec57798ea689
---
 .../core/java/com/android/server/policy/PhoneWindowManager.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/policy/PhoneWindowManager.java b/services/core/java/com/android/server/policy/PhoneWindowManager.java
index ebe8301d018..9da39215604 100644
--- a/services/core/java/com/android/server/policy/PhoneWindowManager.java
+++ b/services/core/java/com/android/server/policy/PhoneWindowManager.java
@@ -7946,7 +7946,7 @@ public class PhoneWindowManager implements WindowManagerPolicy {
 
         synchronized (mLock) {
             int sensorRotation = mOrientationListener.getProposedRotation(); // may be -1
-            if (sensorRotation < 0) {
+            if (sensorRotation < 0 || sensorRotation > 3) {
                 sensorRotation = lastRotation;
             }
 
-- 
2.17.1

